File: VB/MySolution.Module/Employee.vb
Imports System.Linq
Imports System.Text
Imports DevExpress.Persistent.Base
Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel
Imports DevExpress.ExpressApp.DC

Namespace MySolution.Module
    <DefaultClassOptions, ImageName("BO_Employee"), MetadataType(GetType(EmployeeMetadata))> _
    Partial Public Class Employee
    End Class
    Public Class EmployeeMetadata
        Private privateId As Int32
        <Browsable(False)> _
        Public Property Id() As Int32
            Get
                Return privateId
            End Get
            Set(ByVal value As Int32)
                privateId = value
            End Set
        End Property
    End Class
End Namespace

File: CS/MySolution.Module/Employee.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.Persistent.Base;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel;
using DevExpress.ExpressApp.DC;

namespace MySolution.Module {
    [DefaultClassOptions, ImageName("BO_Employee")]
    [MetadataType(typeof(EmployeeMetadata))]
    public partial class Employee {
    }
    public class EmployeeMetadata {
        [Browsable(false)]
        public Int32 Id { get; set; }
    }
}
